app/src/lib/stores/app-store.ts

```diff
@@ -2336,23 +2336,25 @@ export class AppStore extends TypedBaseStore<IAppState> {
     branch: Branch,
     includeRemote: boolean
   ): Promise<void> {
-    return this.withAuthenticatingUser(repository, async (repo, account) => {
-      const defaultBranch = this.getRepositoryState(repository).branchesState
-        .defaultBranch
-      if (!defaultBranch) {
-        throw new Error(`No default branch!`)
+    return this.withAuthenticatingUser(repository, async (r, account) => {
```

Comment:
just to make sure i understand this right—`r` here is going to to be `repository`, correct?

Reply:
`withAuthenticatingUser` will try and update the repository to the latest details before running the callback, so I think unless we have a compelling reason to _not_ do that, we should be using `r`. I just updated this to make the scope distinction more clear.